bot/exts/evergreen/issues.py

```diff
@@ -86,70 +109,85 @@ def check_in_block(message: discord.Message, repo_issue: str) -> bool:
 
     async def fetch_issues(
             self,
-            numbers: set,
+            number: int,
             repository: str,
             user: str
-    ) -> t.Union[FetchIssueErrors, str, list]:
-        """Retrieve issue(s) from a GitHub repository."""
-        links = []
-        if not numbers:
-            return FetchIssueErrors.value_error
-
-        if len(numbers) > MAX_REQUESTS:
-            return FetchIssueErrors.max_requests
-
-        for number in numbers:
-            url = f"https://api.github.com/repos/{user}/{repository}/issues/{number}"
-            merge_url = f"https://api.github.com/repos/{user}/{repository}/pulls/{number}/merge"
-            log.trace(f"Querying GH issues API: {url}")
-            async with self.bot.http_session.get(url, headers=REQUEST_HEADERS) as r:
-                json_data = await r.json()
-
-            if r.status in BAD_RESPONSE:
-                log.warning(f"Received response {r.status} from: {url}")
-                return f"[{str(r.status)}] #{number} {BAD_RESPONSE.get(r.status)}"
-
-            # The initial API request is made to the issues API endpoint, which will return information
-            # if the issue or PR is present. However, the scope of information returned for PRs differs
-            # from issues: if the 'issues' key is present in the response then we can pull the data we
-            # need from the initial API call.
-            if "issues" in json_data.get("html_url"):
+    ) -> t.Union[IssueState, FetchError]:
+        """
+        Retrieve an issue from a GitHub repository.
+
+        Returns IssueState on success, FetchError on failure.
+        """
+        url = ISSUE_ENDPOINT.format(user=user, repository=repository, number=number)
+        merge_url = PR_MERGE_ENDPOINT.format(user=user, repository=repository, number=number)
+        log.trace(f"Querying GH issues API: {url}")
+
+        async with self.bot.http_session.get(url, headers=REQUEST_HEADERS) as r:
+            json_data = await r.json()
+
+        if r.status == 403:
+            if r.headers.get("X-RateLimit-Remaining") == "0":
+                log.info(f"Ratelimit reached while fetching {url}")
+                return FetchError(403, "Ratelimit reached, please retry in a few minutes.")
+            return FetchError(403, "Cannot access issue.")
+        elif r.status in (404, 410):
+            return FetchError(r.status, "Issue not found.")
+        elif r.status != 200:
+            return FetchError(r.status, "Error while fetching issue.")
```

Comment:
You can also mention issue #643 in this PR. As the original contributor who was interested has un-assigned himself and looks no more interested.

Reply:
[`301 Moved Permananently`](https://docs.github.com/en/rest/overview/resources-in-the-rest-api#http-redirects) has still not been handled yet.

Reply:
AFAIK aiohttp will follow the 301 and either yield a 200 or 404.